client/src/pages/Kanban.vue

```diff
@@ -27,91 +27,120 @@
+      const fallbackWipLimit = 20
 
-    function log() {
-      //window.console.log(evt)
-    }
+      let issuesForProject = []
+      let issuesByStatus = ref()
 
-    async function getIssuesForProject(){
-      let response = (await RedmineService.getIssuesForProject(store.state.user.api_key, store.state.query.id, store.state.project.id)).data
-      originalIssues = response.issues
+      function log() {
+        //window.console.log(evt)
+      }
 
-      const uniqueStatusNames = response.issues.reduce((acc, curr) => {
-        return acc.includes(curr.status.name) ? acc : [...acc, curr.status.name]
-      }, []);
+      async function setupColumnConfig() {
+        let redmineStatuses
+        let configIssue
+        let columnNames
+        let wipLimitFromConfig
+        try {
+          redmineStatuses = (await RedmineService.getRedmineStatuses(store.state.user.api_key)).data.issue_statuses
+          configIssue = (await RedmineService.getKanbanConfig(store.state.user.api_key, store.state.project.id)).data.issues[0]
+          let config = JSON.parse(configIssue.description).config
+          columnNames = config.columns || fallbackColumnConfig
+          wipLimitFromConfig = config.WIP || fallbackWipLimit
+        } catch (error) {
+          columnNames = fallbackColumnConfig
+          wipLimitFromConfig = fallbackWipLimit
+        }
+        columnConfig.value = redmineStatuses.filter(status => columnNames.includes(status.name))
+        wipLimit.value = wipLimitFromConfig
+      }
+      async function getIssuesForProject() {
+        issuesForProject.value = (await RedmineService.getIssuesForProject(store.state.user.api_key, store.state.query.id, store.state.project.id)).data.issues
+        issuesByStatus.value = lodash.groupBy(issuesForProject.value, 'status.name')
+      }
 
-      issuesByStatus.value = uniqueStatusNames.map(name => response.issues.filter(i => i.status.name === name));
-      
+      async function add(event){
+        const movedTo = event.to.parentNode.firstElementChild.textContent
+        const movedId = parseInt(event.item.innerText.match(issueIdRegex)[0])
+        const newStatus = columnConfig.value.find(i => i.name === movedTo)
+        const originalIssue = issuesForProject.value.find(i => i.id === movedId)
+        originalIssue.status = newStatus
+        issuesByStatus.value = lodash.groupBy(issuesForProject.value, 'status.name')
+        await RedmineService.updateIssueStatus(store.state.user.api_key, originalIssue.id, newStatus.id)
+      }
 
-      uniqueStatusNamesWithIds = response.issues.reduce((acc, curr) => {
-        return acc.some(i => i.id === curr.status.id) ? acc : [...acc, curr.status]
-      }, []);
-    }
+      function getLimitedList(issueList){
+        if (issueList && issueList.length > wipLimit.value) {
+          return issueList.slice(0, wipLimit.value)
+        } else {
+          return issueList
+        }
+      }
```

Comment:
do we indicate that only a limited number of issue is displayed? usually, the solution i see for wip is that is highlights the column with a red border indicating that the wip limit is surpassed

Reply:
also, in the "Done" column we should be able to have more then 5 elements same as for "New". Now the issues just disappear